Build fails on Mac

This was reported by @chrimaka in https://github.com/dotnet-bluetooth-le/dotnet-bluetooth-le/pull/669#issuecomment-1501657254:

> If I try to build BLE.sln, VS2022 for macOS (17.5.3 (build 15)) already gives me an error during opening  error NETSDK1045: Das aktuelle .NET SDK unterstützt .NET Core 6.0 nicht als Ziel. Geben Sie entweder .NET Core 5.0 oder niedriger als Ziel an, oder verwenden Sie eine .NET SDK-Version, die .NET Core 6.0 unterstützt. Für das Projekt kann keine Paketspezifikation erstellt werden.

I'll note that the build problems do not just affect the build process through VS-for-Mac but also on the command line.

However, the package restoration issues are less severe on the command line, where they only appear as warnings (the affected projects are skipped and the rest of the restoration runs through ok):

```
% dotnet restore Source/BLE.Mac.slnf
~/github/dotnet-bluetooth-le/Source/BLE.Client/BLE.Client.Droid/BLE.Client.Droid.csproj : warning NU1503: Skipping restore for project '~/github/dotnet-bluetooth-le/Source/BLE.Client/BLE.Client.Droid/BLE.Client.Droid.csproj'. The project file may be invalid or missing targets required for restore. [~/github/dotnet-bluetooth-le/Source/BLE.Mac.slnf]
~/github/dotnet-bluetooth-le/Source/BLE.Client/BLE.Client.macOS/BLE.Client.macOS.csproj : warning NU1503: Skipping restore for project '~/github/dotnet-bluetooth-le/Source/BLE.Client/BLE.Client.macOS/BLE.Client.macOS.csproj'. The project file may be invalid or missing targets required for restore. [~/github/dotnet-bluetooth-le/Source/BLE.Mac.slnf]
  Determining projects to restore...
  All projects are up-to-date for restore.
```

But then, when building the solution (or just the Plugin.BLE project) one sees the following errors:

```
% dotnet build Source/Plugin.BLE/Plugin.BLE.csproj
MSBuild version 17.5.1+f6fdcf537 for .NET
  Determining projects to restore...
  All projects are up-to-date for restore.
  Plugin.BLE -> ~/github/dotnet-bluetooth-le/Source/Plugin.BLE/bin/Debug/netstandard2.0/Plugin.BLE.dll
~/.nuget/packages/msbuild.sdk.extras/3.0.44/Build/Workarounds.targets(27,5): error : If you are building projects that require targets from full MSBuild or MSBuildFrameworkToolsPath, you need to use desktop msbuild ('msbuild.exe') instead of 'dotnet build' or 'dotnet msbuild' [~/github/dotnet-bluetooth-le/Source/Plugin.BLE/Plugin.BLE.csproj::TargetFramework=MonoAndroid10.0]
~/.nuget/packages/msbuild.sdk.extras/3.0.44/Build/Workarounds.targets(27,5): error : If you are building projects that require targets from full MSBuild or MSBuildFrameworkToolsPath, you need to use desktop msbuild ('msbuild.exe') instead of 'dotnet build' or 'dotnet msbuild' [~/github/dotnet-bluetooth-le/Source/Plugin.BLE/Plugin.BLE.csproj::TargetFramework=Xamarin.iOS10]
~/.nuget/packages/msbuild.sdk.extras/3.0.44/Build/Workarounds.targets(27,5): error : If you are building projects that require targets from full MSBuild or MSBuildFrameworkToolsPath, you need to use desktop msbuild ('msbuild.exe') instead of 'dotnet build' or 'dotnet msbuild' [~/github/dotnet-bluetooth-le/Source/Plugin.BLE/Plugin.BLE.csproj::TargetFramework=Xamarin.Mac20]
  Plugin.BLE -> ~/github/dotnet-bluetooth-le/Source/Plugin.BLE/bin/Debug/net6.0-android/Plugin.BLE.dll
  Plugin.BLE -> ~/github/dotnet-bluetooth-le/Source/Plugin.BLE/bin/Debug/net6.0-ios/Plugin.BLE.dll
  Plugin.BLE -> ~/github/dotnet-bluetooth-le/Source/Plugin.BLE/bin/Debug/net6.0-maccatalyst/Plugin.BLE.dll
  Plugin.BLE -> ~/github/dotnet-bluetooth-le/Source/Plugin.BLE/bin/Debug/net7.0-android/Plugin.BLE.dll
  Plugin.BLE -> ~/github/dotnet-bluetooth-le/Source/Plugin.BLE/bin/Debug/net7.0-ios/Plugin.BLE.dll
  Plugin.BLE -> ~/github/dotnet-bluetooth-le/Source/Plugin.BLE/bin/Debug/net7.0-maccatalyst/Plugin.BLE.dll

Build FAILED.
```

--- a/.github/workflows/dotnet.yml
+++ b/.github/workflows/dotnet.yml
@@ -9,7 +9,8 @@ on:
   workflow_dispatch:
 
 jobs:
-  build:
+
+  winBuild:
     runs-on: windows-latest
     steps:
     - uses: actions/checkout@v3
@@ -67,3 +68,24 @@ jobs:
       with:
         name: nupkg
         path: ./Source/*/nuget/*Plugin.BLE*.nupkg
+
+  macBuild:
+    runs-on: macos-latest
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+    - uses: nuget/setup-nuget@v1
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v3
+      with:
+        dotnet-version: 7.0.x
+    - name: Install .NET MAUI
+      run: |
+        dotnet nuget locals all --clear
+        dotnet workload install maui --source https://aka.ms/dotnet6/nuget/index.json --source https://api.nuget.org/v3/index.json
+        dotnet workload install android ios maccatalyst tvos macos maui wasm-tools maui-maccatalyst --source https://aka.ms/dotnet6/nuget/index.json --source https://api.nuget.org/v3/index.json
+    - name: Build Plugin.BLE NuGet
+      run: dotnet build ./Source/Plugin.BLE/Plugin.BLE.csproj /p:Configuration=Release /t:restore,build,pack /p:Version=$(git describe) /p:ContinuousIntegrationBuild=true /p:DeterministicSourcePaths=false
+    - name: Build MVVMCross.Plugins.BLE NuGet
+      run: dotnet build ./Source/MvvmCross.Plugins.BLE/MvvmCross.Plugins.BLE.csproj /p:Configuration=Release /t:restore,build,pack /p:Version=$(git describe) /p:ContinuousIntegrationBuild=true /p:DeterministicSourcePaths=false

--- a/Source/BLE.Mac.slnf
+++ b/Source/BLE.Mac.slnf
@@ -5,10 +5,7 @@
       "Plugin.BLE\\Plugin.BLE.csproj",
       "Plugin.BLE.Tests\\Plugin.BLE.Tests.csproj",
       "MvvmCross.Plugins.BLE\\MvvmCross.Plugins.BLE.csproj",
-      "BLE.Client\\BLE.Client\\BLE.Client.csproj",
-      "BLE.Client\\BLE.Client.Droid\\BLE.Client.Droid.csproj",
-      "BLE.Client\\BLE.Client.iOS\\BLE.Client.iOS.csproj",
-      "BLE.Client\\BLE.Client.macOS\\BLE.Client.macOS.csproj"
+      "BLE.Client\\BLE.Client\\BLE.Client.csproj"
     ]
   }
 }

--- a/Source/MvvmCross.Plugins.BLE/MvvmCross.Plugins.BLE.csproj
+++ b/Source/MvvmCross.Plugins.BLE/MvvmCross.Plugins.BLE.csproj
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="MSBuild.Sdk.Extras/3.0.44">
 	<PropertyGroup>
-		<TargetFrameworks>MonoAndroid10.0;Xamarin.iOS10;Xamarin.Mac20;net6.0-android;net6.0-ios;net6.0-maccatalyst;net7.0-android;net7.0-ios;net7.0-maccatalyst</TargetFrameworks>
-		<TargetFrameworks Condition=" '$(OS)' == 'Windows_NT' ">$(TargetFrameworks);uap10.0.19041;net6.0-windows10.0.19041;net7.0-windows10.0.19041</TargetFrameworks>
+		<TargetFrameworks>net6.0-android;net6.0-ios;net6.0-maccatalyst;net7.0-android;net7.0-ios;net7.0-maccatalyst</TargetFrameworks>
+		<TargetFrameworks Condition=" '$(OS)' == 'Windows_NT' ">$(TargetFrameworks);MonoAndroid10.0;Xamarin.iOS10;Xamarin.Mac20;uap10.0.19041;net6.0-windows10.0.19041;net7.0-windows10.0.19041</TargetFrameworks>
 		<AssemblyName>MVVMCross.Plugins.BLE</AssemblyName>
 		<RootNamespace>MVVMCross.Plugins.BLE</RootNamespace>
 		<Version>3.0.0-beta.3</Version>

--- a/Source/Plugin.BLE/Plugin.BLE.csproj
+++ b/Source/Plugin.BLE/Plugin.BLE.csproj
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="MSBuild.Sdk.Extras/3.0.44">
 	<PropertyGroup>
-		<TargetFrameworks>netstandard2.0;MonoAndroid10.0;Xamarin.iOS10;Xamarin.Mac20;net6.0-android;net6.0-ios;net6.0-maccatalyst;net7.0-android;net7.0-ios;net7.0-maccatalyst</TargetFrameworks>
-		<TargetFrameworks Condition=" '$(OS)' == 'Windows_NT' ">$(TargetFrameworks);uap10.0.19041;net6.0-windows10.0.19041;net7.0-windows10.0.19041</TargetFrameworks>
+		<TargetFrameworks>netstandard2.0;net6.0-android;net6.0-ios;net6.0-maccatalyst;net7.0-android;net7.0-ios;net7.0-maccatalyst</TargetFrameworks>
+		<TargetFrameworks Condition=" '$(OS)' == 'Windows_NT' ">$(TargetFrameworks);MonoAndroid10.0;Xamarin.iOS10;Xamarin.Mac20;uap10.0.19041;net6.0-windows10.0.19041;net7.0-windows10.0.19041</TargetFrameworks>
 		<AssemblyName>Plugin.BLE</AssemblyName>
 		<RootNamespace>Plugin.BLE</RootNamespace>
 		<Version>3.0.0-beta.3</Version>

